heat/core/tensor.py

```diff
@@ -29,17 +30,38 @@ def dtype(self):
     def gshape(self):
         return self.__gshape
 
+    @property
+    def size(self):
+        try:
+            return reduce(lambda i, j: i * j, self.array_shape)
+        except TypeError:
+            # todo: throw a soft warning about taking the number of elements of a single element tensor
+            return 1
+
+    @property
+    def gnumel(self):
+        return self.size
+
+    @property
+    def lnumel(self):
+        return reduce(lambda i, j: i * j, self.lshape)
```

Comment:
code duplication with line 36. Proposal: introduce a prod(<iterable>) function implemented as you did or via

```
import operator
def prod(iterable)
    reduce(operator.mul, iterable, 1)
```

Reply:
what about numpy.prod?

Reply:
Sure would work as well